src/ext/spec/item.cr

```diff
@@ -0,0 +1,12 @@
+module Spec
+  module Item
+    def all_tags : Set(String)
+      all_tags = tags || Set(String).new
+      temp = parent
```

Comment:
Does this include all parents, or just the direct parent? 

```crystal
describe Thing do
  context "when thing", tags: ["chrome"] do
    describe "is a thing" do
      it "does stuff" do
      end
    end
  end
end
```

Reply:
Yes, in the `if` below I call `all_tags` on the parent which means it will fetch all tags of ever parent until it reaches the `Spec::RootContext` which is not a `Spec::Item`

Reply:
ah, ok. I see what it's doing now. Can we rename the internal variable so it doesn't shadow the method name?

Reply:
I ended up using the same naming convention we were recommended to use over here https://github.com/luckyframework/avram/blob/ac281f490a85283eaf8b74fd23357dd70a6d7f22/src/ext/db/connection.cr#L7

with an underscore and the name of the shard so in this case `_lucky_flow_all_tags`. I also added `# :nodoc:`. It's not a very clear name but it doesn't need to be

Reply:
Cool, I like that.